src/runtime/Codecs/FunctionCodec.cs

```diff
@@ -0,0 +1,44 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace Python.Runtime.Codecs
+{
+    //converts python functions to C# actions
+    class FunctionCodec : IPyObjectDecoder
+    {
+        public static FunctionCodec Instance { get; } = new FunctionCodec();
+        public bool CanDecode(PyObject objectType, Type targetType)
+        {
+            if (!objectType.IsCallable()) return false;
+
+            //TODO - handle nonzero arguments.
+            var args = targetType.GetGenericArguments();
```

Comment:
Before trying anything else, you should check if your codec actually supports `targetType` too.